packages/core/core/src/types.ts

```diff
@@ -582,3 +582,38 @@ export interface Context {
 export interface FactoryParams {
   provide?: (context: Context) => any;
 }
+
+export interface ApiClientExtensionLifecycle {
+  beforeCreate?: (config, headers?: Record<string, string>) => any;
```

Comment:
same as below also can we make this just:
```suggestion
  beforeCreate?: ({ config, headers: Record<string, string> }) => any;
```
for maintainability?